atd-vzv/src/views/map/Map.js

```diff
@@ -257,30 +233,20 @@ const Map = () => {
       {buildHighInjuryLayer(overlay)}
       {!!cityCouncilOverlay && overlay.name === "cityCouncil" && (
         <Source type="geojson" data={cityCouncilOverlay}>
-          {/* Add beforeId to render beneath crash points */}
-          <Layer beforeId="base-layer" {...cityCouncilDataLayer} />
+          {/* Add beforeId to render beneath crash points, road layer, and map labels */}
+          <Layer beforeId="road-street" {...cityCouncilDataLayer} />
         </Source>
       )}
-      {/* Render crash point tooltip */}
+      {/* Render feature info or popup */}
       {selectedFeature && (
-        <MapCrashInfoBox
+        <MapInfoBox
           selectedFeature={selectedFeature}
           setSelectedFeature={setSelectedFeature}
           isMobile={isMobile}
+          type={selectedFeature.layer.id}
         />
       )}
-      {/* Show spinner when map is updating */}
-      {isMapDataLoading && (
-        <StyledMapSpinner className="fa-layers fa-fw">
-          <FontAwesomeIcon icon={faCircle} color={colors.infoDark} size="4x" />
-          <FontAwesomeIcon
-            className="needle"
-            icon={faCompass}
-            color={colors.dark}
-            size="4x"
-          />
-        </StyledMapSpinner>
-      )}
+      <MapCompassSpinner isSpinning={isMapDataLoading} />
```

Comment:
love it when a new stateless functional component is born.

Reply:
Me too! ⚛️👶🍼